src/utils/yaml-utils.ts

```diff
@@ -6,7 +6,12 @@ import { sanitizer } from "@services/utilServices/Sanitizer"
 // else in the codebase.
 export const sanitizedYamlParse = (
   unparsedContent: string
-): Record<string, unknown> => yaml.parse(sanitizer.sanitize(unparsedContent))
+): Record<string, unknown> =>
+  yaml.parse(unparsedContent, (key, value) =>
+    typeof value === "string" ? sanitizer.sanitize(value) : value
+  )
```

Comment:
Just checking, wouldn't this mean that any nested fields (e.g. 
```
operating_hours:
      - days: aa
        time: "12"
        description: "123"
      - days: bb
        time: "13"
        description: <script>alert('hi')</script>
```

would end up unsanitised? It gets blocked by our csp right now, but the page itself is able to save this without sanitisation

Reply:
since the doc for yaml itself points you to the `json.parse` [page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#using_the_reviver_parameter), i'm just gonna paste the ss here to save you the effort HAHA

![Screenshot 2023-04-06 at 1 25 10 PM](https://user-images.githubusercontent.com/44049504/230279510-1642b0fb-b316-4885-a611-eefe92c9b207.png)

the effect of this means that only **leaf nodes** of type `string` gets sanitized and we leave safe terminal values of other types (yaml has `null | boolean | number | string` for scalar values) untouched

Reply:
turns out this affects `contact-us` and more due to it calling the v1 endpoint to update pages. this isn't caused by this PR and has been found to work on prod (albeit no impact due to CSP blocking). 

filed as IS-75 due i'll be resolving this as it's out of scope